x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.tsx

```diff
@@ -187,272 +187,275 @@ export function LayerPanel(
+                                    prevState: layerDatasourceState,
+                                  }),
+                                  activeVisualization.removeDimension({
+                                    layerId,
+                                    columnId: id,
+                                    prevState: props.visualizationState,
+                                  })
+                                );
+                                removeButtonRef(id);
+                              }}
+                            >
+                              <NativeRenderer
+                                render={layerDatasource.renderDimensionTrigger}
+                                nativeProps={{
+                                  ...layerDatasourceConfigProps,
+                                  columnId: accessorConfig.columnId,
+                                  filterOperations: group.filterOperations,
+                                }}
+                              />
+                            </DimensionButton>
+                          </div>
+                        </DraggableDimensionButton>
+                      );
+                    })}
+                  </ReorderProvider>
+                  {group.supportsMoreColumns ? (
+                    <EmptyDimensionButton
+                      group={group}
+                      groupIndex={groupIndex}
+                      layerId={layerId}
+                      layerIndex={layerIndex}
+                      layerDatasource={layerDatasource}
+                      layerDatasourceDropProps={layerDatasourceDropProps}
+                      onClick={(id) => {
+                        setActiveDimension({
+                          activeGroup: group,
+                          activeId: id,
+                          isNew: true,
+                        });
                       }}
+                      onDrop={onDrop}
                     />
-                  </div>
-                )}
-            </>
-          }
-        />
+                  ) : null}
+                </>
+              </EuiFormRow>
+            );
+          })}
 
-        <EuiSpacer size="m" />
+          <EuiSpacer size="m" />
 
-        <EuiFlexGroup justifyContent="center">
-          <EuiFlexItem grow={false}>
-            <RemoveLayerButton
-              onRemoveLayer={onRemoveLayer}
-              layerIndex={layerIndex}
-              isOnlyLayer={isOnlyLayer}
-            />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiPanel>
-    </section>
+          <EuiFlexGroup justifyContent="center">
+            <EuiFlexItem grow={false}>
+              <RemoveLayerButton
+                onRemoveLayer={onRemoveLayer}
+                layerIndex={layerIndex}
+                isOnlyLayer={isOnlyLayer}
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiPanel>
+      </section>
+
+      <DimensionContainer
```

Comment:
For reviewers: the basic change is to extract this `DimensionContainer` component out of its previous location inside the `section` that we track by reference. I made this change by inspecting the EUI focus/blur events and noticing that they had a lot of shared parents, and that this was not needed in my local test case with EUI forms.